### ISSUE TITLE
Rule proposal: disallow `window.document`/`window.location`/`window.localStorage` in favour of `document`/`location`/`localStorage`

### ISSUE THREAD

**Issue by dimaMachina** [sindresorhus/eslint-plugin-unicorn#1959]
### Description

since `document`/`location`/`localStorage` are global variables calling `window.*` is redundant

### Fail

```js
window.localStorage.getItem('foo')
```

### Pass

```js
localStorage.getItem('foo')
```

**Comment by fisker** [comment]
Personally, I prefer only use ES globals, I even do `globalThis.setTimeout(...)`. Anyway I didn't strictly follow this rule nowadays, eg I use `document` directly sometime.

**Comment by B2o5T** [comment]
@fisker The generic rule for allowing/disallowing `globalThis`/`window` for global variables will satisfy both cases

**Comment by B2o5T** [comment]
> It feels like this was already suggested maybe on eslint's repo, I can't find the issue here.

even it was already suggested they will reject this rule since it's stylistic rule

**Comment by silverwind** [comment]
Adding `window` to globals can [break stuff with dumb bundlers](https://github.com/vitejs/vite/issues/9901), thought that's 100% the bundler's fault of course.

But I do see value in enforcing `globalThis` or `window` on certain names so that one does not accidentially use the global when they meant to use a local variable, and I do emulate this to a degree in my config via:

```yaml
no-restricted-globals: [2, addEventListener, blur, close, closed, confirm, defaultStatus, defaultstatus, error, event, external, find, focus, frameElement, frames, history, innerHeight, innerWidth, isFinite, isNaN, length, location, locationbar, menubar, moveBy, moveTo, name, onblur, onerror, onfocus, onload, onresize, onunload, open, opener, opera, outerHeight, outerWidth, pageXOffset, pageYOffset, parent, print, removeEventListener, resizeBy, resizeTo, screen, screenLeft, screenTop, screenX, screenY, scroll, scrollbars, self, scrollBy, scrollTo, scrollX, scrollY, status, statusbar, stop, toolbar, top]

**Comment by fregante** [comment]
Closing in favor of broader rules:

- https://github.com/sindresorhus/eslint-plugin-unicorn/issues/2419
- #978
